pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -433,23 +440,85 @@ cdef class {{name}}HashTable(HashTable):
                                                  "Vector.resize() needed")
                             uniques.resize()
                     append_data_{{dtype}}(ud, val)
-                    labels[i] = count
-                    count += 1
+                    if return_inverse:
+                        self.table.vals[k] = count
+                        labels[i] = count
+                        count += 1
+                elif return_inverse:
```

Comment:
my point was just always construct the inverse, which is just the factorization, and simply leave this as _factorize as that is what it is. Channing the names and conditionally having return_inverse are just not worth the added complexity. I can live with a small perf hit when only uniques are needed. 

i also think we can simply remove na_sentinel as its always -1 (but can do that separately).

Reply:
You will see that the block above (previously lines 419-422) got shifted down here verbatim -- that part is only necessary if an inverse is required, and we can avoid more complicated if-conditions (or even checking the bint `return_inverse`) if the case `k == self.table.n_buckets` is tested first.

Might be more clear if you just expand the unchanged lines in between (it's only 3 lines that got hidden)